pennylane/data/data_manager/__init__.py

```diff
@@ -423,48 +451,98 @@ def list_attributes(data_name):
-        raise ValueError(f"Must enter a list of integers between 1 and {len(options)}") from e
-    if any(choice < 1 or choice > len(options) for choice in choices):
-        raise ValueError(f"Must enter a list of integers between 1 and {len(options)}")
-    return [options[choice - 1] for choice in choices]
-
-
-def _interactive_request_single(node, param):
-    """Prompt the user to select a single option from a list."""
-    options = list(node)
-    if len(options) == 1:
-        print(f"Using {options[0]} as it is the only {param} available.")
-        sleep(1)
-        return options[0]
-    print(f"Please select a {param}:")
-    print("\n".join(f"\t{i+1}) {option}" for i, option in enumerate(options)))
-    try:
-        choice = int(input(f"Choice [1-{len(options)}]: "))
-    except ValueError as e:
-        raise ValueError(f"Must enter an integer between 1 and {len(options)}") from e
-    if choice < 1 or choice > len(options):
-        raise ValueError(f"Must enter an integer between 1 and {len(options)}")
-    return options[choice - 1]
+    print(
+        'Please select a list of attributes from the following available attributes or "full" for all attributes.'
+    )
+    for i, option in enumerate(attribute_options):
+        print(f"{i + 1}: {option}")
+    choice = input("Choice of attributes: ")
+    if choice == "full":
+        return attribute_options
+    if not set(choice).issubset(set(attribute_options)):
+        raise ValueError(f"Must select a list of attributes from {attribute_options}")
+    return choice
+
+
+def _interactive_requests(parameters, parameter_tree):
+    """Prompts the user to select parameters for datasets one at a time."""
+
+    branch = parameter_tree
+    choices = []
+    for param in parameters:
+        print(f"Available options for {param}:")
+        for i, option in enumerate(branch["next"].keys()):
+            print(f"{i + 1}: {option}")
+        user_value = input(f"Please select a {param}:")
+        try:
+            branch = branch["next"][user_value]
+        except KeyError as e:
+            raise ValueError(f"Must enter a valid {param}:") from e
+        choices.append(user_value)
+        if "next" in branch:
+            if len(branch["next"]) == 1:
+                branch = next(iter(branch["next"].values()))
+            continue
+        branch = branch["value"]
+        return branch
+
+
+def _get_parameter_tree(class_id) -> tuple[list[str], list[str], dict]:
+    """Returns the (parameters, attributes, parameter_tree) for a given ``class_id``."""
+
+    response = _get_graphql(
+        GRAPHQL_URL,
+        """
+        query GetParameterTree($datasetClassId: String!) {
+          datasetClass(id: $datasetClassId) {
+            attributes {
+              name
+            }
+            parameters {
+              name
+            }
+            parameterTree
+          }
+        }
+        """,
+        {"input": {"datasetClassId": class_id}},
```

Comment:
```suggestion
        {"datasetClassId": class_id}
```
The keys of `variables` will always correspond to the arguments to the query